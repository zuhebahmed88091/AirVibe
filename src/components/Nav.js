import React from 'react';
import PropTypes from 'prop-types';
import logo from './Assets/weather-app.png';
import continents from './Assets/continents.png';
import Filter from './Filter';

const Nav = ({ handleFilterChange }) => (
  <>
    <nav className="navbar">
      <img className="logo" src={logo} alt="logo" />
      <h1>AirVibe</h1>
      <Filter handleFilterChange={handleFilterChange} />
    </nav>
    <header className="map-header">
      <div className="map">
        <img src={continents} alt="Asia Map" />
      </div>
      <h1>Air Quality in Global Cities</h1>
    </header>
  </>
);

Nav.propTypes = {
  handleFilterChange: PropTypes.func.isRequired,
};

export default Nav;
