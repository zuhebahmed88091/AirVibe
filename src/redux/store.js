import { configureStore } from '@reduxjs/toolkit';
import weatherReducer from './weatherSlice';

const store = configureStore({
  reducer: weatherReducer,
});

export default store;
