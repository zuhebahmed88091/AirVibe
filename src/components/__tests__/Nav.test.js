import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Nav from '../Nav';

const customRender = (component) => render(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('Nav', () => {
  it('render Nav Components', () => {
    customRender(<Nav />);
    expect(Nav).toMatchSnapshot();
  });
});
