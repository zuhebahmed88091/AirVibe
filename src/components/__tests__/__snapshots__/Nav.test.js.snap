// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav render Nav Components 1`] = `[Function]`;
